CRM/Core/BAO/CustomGroup.php

```diff
@@ -1832,11 +1832,15 @@ public static function formatGroupTree(&$groupTree, $groupCount = 1, &$form = NU
     $formattedGroupTree = array();
     $uploadNames = $formValues = array();
 
-    // JSON encoded submitted form values passed via URL
-    $submittedValues = CRM_Utils_Request::retrieve('post', 'String');
-    if (!empty($submittedValues)) {
-      $submittedValues = json_decode($submittedValues, TRUE);
-    }
+    // Generate cache key against which submitted custom value will be stored
+    $cacheKey = implode('_',
+      array(__CLASS__,
```

Comment:
`__CLASS__` will always be the same e.g. for forms open in multiple tabs. Do we have access to qfKey here?